crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -143,6 +152,7 @@ impl RuffSettingsIndex {
                                 directory.to_path_buf(),
                                 Arc::new(RuffSettings {
                                     path: Some(pyproject),
+                                    unsafe_fixes: settings.unsafe_fixes,
```

Comment:
Do you remember what's the reason for only storing parts of the settings instead of the entire `settings` struct with accessors for the linter/formatter settings?

Reply:
I think it started out with just having `LinterSettings` and `FormatterSettings` but later on we realized (via bug reports) that exclusions weren't being considered by the server which led to adding the `FileResolverSettings`.

> It makes me wonder if we should simply store the entire `Settings` struct in `RuffSettings`. It seems like we eventually will need all of them anyway?

Yeah, I agree. I can do that as a follow-up.